docs/source/how-to/authentication.md

```diff
@@ -0,0 +1,168 @@
+prompt you again if you log out, or when you session eventually expires or
+is revoked by the server.
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...")
+# Automatically logged in!
+
+# This is a quick way to verify whether you are already logged in
+>>> client.context
+<Context authenticated as '...'>
+```
+
+To opt out of this for subsequent connections, set `remember_me=False`:
+
+```py
+>>> from tiled.client import from_uri
+>>> client = from_uri("https://...", remember_me=False)
+
+````{note}
+Tiled stores OAuth2 tokens (it _never_ stores your password) in files
+with properly restricted permissions under `$XDG_CACHE_DIR/tiled/tokens`.
+The path used by your client can be found at:
+
+  ```py
+  >>> client.context.http_client.auth.token_directory
+  ```
+
+To customize the location of this storage, set the environment variable
+`TILED_CACHE_DIR`.
+````
+
+In the rare where a session is revoke or expired during active use,
+a `tiled.client.auth.CannotRefreshAuthentication` exception is raised.
+Use `client.login()`, as directed by the error message.
+
+## Noninteractive Authentication (API keys)
+
+There are environments where logging in interactively is not possible,
+such as running a batch script. For these applications, we recommend
+using an API key. These can be created from the CLI:
+
+```sh
+$ tiled login
+$ tiled api_key create --expires-in 7d --note "for this week's experiment"
+```
+
+or from an interactive Python session:
+
+```py
+>>> client = from_uri("https://...")
+>>> client.login()
+>>> client.create_api_key(expires_in="7d", note="for this week's experiment")
+{"secret": ...}
+```
+
+The expiration and note are optional, but recommended. Expiration can be given
+in units of years `y`, days `d`, hours `h`, minutes `m`, or seconds `s`.
+
+The best way to provide an API key is to set the environment variable
+`TILED_API_KEY`. A script like this:
+
+```py
+from tiled.client import from_uri
+
+client = from_uri("https://....")
+```
+
+will detect that `TILED_API_KEY` is set and use that API key for
+authentication with Tiled. This is equivalent to:
+
+```py
+import os
+from tiled.client import from_uri
+
+client = from_uri("https://....", api_key=os.environ["TILED_API_KEY"])
+```
+
+Avoid typing the API key in to the code:
+
+```py
+from_uri("https://...", api_key="secret!")  # DON'T
+```
+
+as it is easy to accidentally share or leak.
+
+## Custom Applications
+
+Custom applications, such as a graphical interfaces that wrap Tiled,
+may not be able to use Tiled commandline-based prompts. They may
+provide their own mechanism for collection credentials
+(for password grants) or launching a browser and waiting for the
+user to authorize a session (for device code grants).
+
+Custom applications should avoid using the convenience functions
+`tiled.client.constructors.from_uri` and
+`tiled.client.constructors.from_profile`. Instead, follow this pattern:
+
```

Comment:
This might be a good place to link: https://github.com/bluesky/tiled/blob/799f7ba163a9424a44a366e232fc995854d12470/tiled/client/context.py#L55-L95

as a sensible default.